servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java

```diff
@@ -496,15 +506,34 @@ private static Reference makeRef(NamedReference<CommitMeta> refWithHash) {
           .hash(refWithHash.getHash().asString())
           .build();
     } else if (ref instanceof BranchName) {
-      return ImmutableBranch.builder()
-          .name(ref.getName())
-          .hash(refWithHash.getHash().asString())
-          .build();
+      Builder builder =
+          ImmutableBranch.builder().name(ref.getName()).hash(refWithHash.getHash().asString());
+      Map<String, Object> metadataProperties = extractMetadataProperties(refWithHash);
+      if (!metadataProperties.isEmpty()) {
+        builder.metadataProperties(metadataProperties);
+      }
+      return builder.build();
     } else {
       throw new UnsupportedOperationException("only converting tags or branches"); // todo
     }
   }
 
+  private static Map<String, Object> extractMetadataProperties(
+      NamedReference<CommitMeta> refWithHash) {
+    Map<String, Object> metadataProperties = new HashMap<>();
+    if (null != refWithHash.getAheadBehind()) {
+      metadataProperties.put("numCommitsAhead", refWithHash.getAheadBehind().getAhead());
```

Comment:
Need contstants for these - and have the keys documented in the REST API, with an example for each.